metrics/docs/02_sign_and_notarize_metrics_binaries.md

```diff
@@ -0,0 +1,176 @@
+# Motivation
+> What problem is this project solving?
+
+Apple's ongoing initiatives at controlling what runs on their platforms took a new turn with macOS Catalina (10.15), with required app and command-line binary signing.
+
+[Codesign and notarize](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution) enables programs downloaded from the Internet to be opened without any warnings.
+
+To solve the described issue, the following steps is required:
+ - signing the binary with a Developer ID Certificate
+ - notarization (uploading the binary to Apple for approval)
+ - validating code signing
+ - validating notarization
+
+Currently, Metrics binaries for macOS have not signed or notarized.
+
+The document's goal is to investigate and analyze tools and methods that can help in signing and notarizing binaries.
+
+# References
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [Notarizing macOS Software Before Distribution](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution)
+- [code signing for macOS](https://wiki.freepascal.org/Code_Signing_for_macOS)
+- [Using app-specific passwords](https://support.apple.com/en-us/HT204397)
+- [gon](https://github.com/mitchellh/gon)
+- [signing_tools](https://github.com/drud/signing_tools)
+- [notarize-cli](https://github.com/bacongravy/notarize-cli)
+- [Import codesign certificate Github Action](https://github.com/Apple-Actions/import-codesign-certs)
+
+# Analyze
+
+- [Requirements](#requirements)
+- [Analyzing process](#analyzing-process)
+- [Decision](#decision)
```

Comment:
```suggestion
```

I'm not sure if we need to have such TOC at the beginning of the large section. While it could improve navigation I'm not sure that the structure should be like that still.